test/test_expand.jl

```diff
@@ -26,6 +26,9 @@ using Test
     @test isequal(qexpand((B+C+D)*A), B*A + C*A + D*A)
     @test isequal(qexpand(commutator(A, B) * C), A*B*C - B*A*C)
 
+    @test isequal(qexpand(A*(B⊗C⊗D)), (A*B)⊗(A*C)⊗(A*D))
+    @test isequal(qexpand((B⊗C⊗D)*A), (B*A)⊗(C*A)⊗(D*A))
```

Comment:
These two seem wrong.

For `A*(B⊗C⊗D)` one must have `basis(A)==basis(B⊗C⊗D)`.

For `(A*B)⊗(A*C)⊗(A*D)` one must have `basis(A)==basis(B)==basis(C)==basis(D)`.

We should probably have a basis consistency check in the operator multiplication function (and we can add such checks in the apply functions as well).

Reply:
I was thinking about this issue when I was writing docs earlier today. I suppose we would also want to have a consistency check in our definition for `isequal`?